backend/controllers/serviceexport/serviceexport_controller.go

```diff
@@ -128,8 +130,16 @@ func (r *APIServiceExportReconciler) Reconcile(ctx context.Context, req mcreconc
 		return ctrl.Result{}, err
 	}
 
-	// Update status if it has changed
-	if !equality.Semantic.DeepEqual(original, apiServiceExport) {
+	// Update annotatations changed (hash)
+	if !equality.Semantic.DeepEqual(original.Annotations, apiServiceExport.Annotations) {
+		err := client.Update(ctx, apiServiceExport)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update APIServiceExport status: %w", err)
```

Comment:
This is not actually updating the status, so either the code or the error message are off.

Reply:
Error. Adding better comment too.